### DISCUSSION TITLE
Problem with the pipeline

### DISCUSSION THREAD

**Discussion by wdkrnls** [ropensci/targets#898] [Help]
Hi,

Do you have any suggestions for debugging a situation where none of the debugging documentation is applicable yet because the targets graph itself failed to build?

I have been thinking about this problem for a few weeks, but it's frustratingly non-deterministic. Let me explain.

My targets pipeline ran fine in June, but stopped working in July when I came back to run it again. The error is:

```
Error in graph_edges(edges_envir(envir)) :
  igraph::is_dag(graph) is not TRUE
X Problem with the pipeline
```

What I have observed is that once any pipeline fails to build in an R process, nothing works after that until I restart the process. However, I have not found any shareable code which could repeat the situation. Have you seen anything like this?

FYI, no code in my workflow changed, but I did update igraph from the 1.2.x series to the 1.3.x series.

**Comment by wlandau** [comment]
If restarting your R session resolves the issue temporarily, maybe it's a global object you define `_targets.R` that somehow creates the circularity. Loops in the graph are tricky to pick out, but you could try iteratively commenting out sections of the pipeline and seeing if that allows you to generate the graph, and if so what the graph looks like.

**Reply by wlandau** [reply]
What I'm referring to is the fact that `targets` thinks your pipeline has a circular relationship somewhere: for example, maybe target A depends on target B, target B depends on function C, but maybe function C appears to depend on target A. A --> B --> C --> A is not a directed acyclic graph because it contains a cycle. It will probably require experimentation and isolating pieces of the pipeline to figure out where the cycle is.

**Comment by wdkrnls** [comment]
Thanks for your suggestion. That makes sense and also suggested to me that maybe it was possible that the graph edges might not be satisfiable if there was a typo in a dependency name.  However, I copied my code line by line into another targets.R file, checking after each whether it still worked. I was able to get the whole pipeline to run this way, changing only a few cue arguments along the way from "never" to "thorough". Thankfully this pipeline only takes a little over an hour to build. Thinking I had figured it out, I went back to the original script and made those changes to the cues. Yet the pipeline still had the is_dag error above. I thought maybe last time was a fluke, so I restarted R and tried again. The new workflow script still worked. The old one still failed. I used a visual_diff tool to look for differences. Only the indentation and position of the parentheses were obviously different. I will try tomorrow to see if I can bring the new script to character parity and see if that reveals the real issue.

**Reply by wdkrnls** [reply]
Coming at the code fresh I did see that this line somehow didn't make it to the new version:

```
tar_option_set(imports = internal_pkgs, workspace_on_error = TRUE)
```

The old code runs when I delete it. I had `library(pkgN, quietly = TRUE)`  calls in both versions.
